### ISSUE TITLE
Bird Buddy stops running. 

### ISSUE THREAD

**Issue by RuudvanMunster** [jhansche/ha-birdbuddy#73]
It looks like Bird Buddy integration has stopped running. Since October 20, it's entities are not updated anymore. If I restart HA the BB entities are updated once and I get the following message. 
My first impression is that it is since I updated to HA 2023.10. But I am not really sure about that. My BB has been low on battery. But I did several restarts of HA since I recharged the BB battery. 

Logger: birdbuddy
Source: custom_components/birdbuddy/sensor.py:201
Integration: Bird Buddy (documentation, issues)
First occurred: 13:00:09 (1 occurrences)
Last logged: 13:00:09

Unexpected Feed type: FeedItemCollectedPostcard

**Comment by jhansche** [comment]
Thanks for the report.

It looks like they've updated the schema by adding a new feed item type, and the library is being strict about what it can interpret. That obviously isn't safe, so we should be able to allow unknown feed item types and just ignore them (or try to interpret them generically)

**Comment by RuudvanMunster** [comment]
Could the ne feed item type have to do soemthing with the new subscription they offered? Part of that is a higher resolution video.

**Comment by jhansche** [comment]
No, the new feed item is not subscription related.

It looks like the new feed item has an expiration time, and includes some metadata like whether it includes mystery visitors, new/unlocked species, and a list of species in the postcard.

The actual problem is that even though the new feed item's schema shows that it should contain a `createdAt` timestamp field, the feed items do not contain data for that field. So trying to convert it to a Python timestamp is failing.

**Comment by jhansche** [comment]
So, the feed item does include the createdAt timestamp - but we weren't selecting it, because we didn't know that it should exist (the downsides of GraphQL).

Testing a fix now, which should not only resolve this issue, but also allow future added feed items to be handled gracefully instead of breaking.

The warning log you mentioned will still continue to be logged in these cases. But that was not what broke the coordinator. What actually broke it was this:
```
ERROR (MainThread) [custom_components.birdbuddy] Error fetching birdbuddy data: object of type 'NoneType' has no len()
```
and that's what I'm fixing now.

**Comment by jhansche** [comment]
The change appears to be working for me. I'll merge it and get it released: jhansche/pybirdbuddy#20

**Comment by RuudvanMunster** [comment]
Great! Thank you.

**Comment by RuudvanMunster** [comment]
Oops. Maybe we should reopen this issue? The problem with the logbook message disappeared indeed. But the problem that the attributes are updated only once (at restart of HA) still persists. 
I also found that a photo of 1 week ago now has been presented as a new one, with the time stamp of the moment I restart HA. The value of the wifi strength and the battery level have the same problem. Reason that I did not see that my BB’s battery level was too low.
I really think this has to do with the change toward the PRO subscription. I think that they did something with resolutions. A video I downloaded before they offered the PRO subscription had a 1080 video resolution. If I download a video now it has the 720 resolution. So they must have reduced the standard resolution and offered the previous standard resolution as a PRO option. As you may expect I have a meaning about that…
My impression is that they have been fooling around with the interface, which may explain the other problems I have now. Maybe they try to block your API?

**Comment by jhansche** [comment]
No, it has nothing to do with subscription, or videos, at all.

After the v0.0.18 update, my integration is now receiving updates as expected, including battery and signal levels; and the automation to collect postcards is working properly now, on its own. I can confirm that the auto-collected postcard is also appearing in my BB app as a visitor, and a separate automation that I have that triggers on `new_postcard` and increments a helper counter was triggered and completed successfully.

If your battery level is too low now, maybe that is the problem? Try charging it first and see if the integration updates to show that it is charging.

**Comment by RuudvanMunster** [comment]
I did already charge the battery. So that should not be the problem. Switching the Power Profile works fine in both directions. I will check the entities and the visitors next morning, as the BB is now in sleeping mode.
I'll let you know the outcome!

**Comment by RuudvanMunster** [comment]
However I still have this problem. The last update of this entity was 42 minutes ago, which is exactly the moment I restarted HA. I still do not understand that.
![image](https://github.com/jhansche/ha-birdbuddy/assets/30747626/885fa1b1-899a-4945-9c31-6ac91f395842)

**Comment by jhansche** [comment]
So if the Power Profile test works, then that should show that the coordinator is updating on its own. If it's in sleeping mode, then it probably won't update, at least not as frequently. Looking back at historical graphs on my end, I see 2 things:
1. While the feeder is "Sleeping", the updates are happening less frequently. When it's "Ready", the updates happen more often.
2. The issue began Oct 19 at 8am (US EDT).

![Screenshot 2023-10-23 at 3 07 33 PM](https://github.com/jhansche/ha-birdbuddy/assets/99272/d1a9d456-982f-483b-9008-10520cb84698)

**Comment by RuudvanMunster** [comment]
I had the problem of my entities not updating also at day time. 
![image](https://github.com/jhansche/ha-birdbuddy/assets/30747626/721dbf77-ba7c-4708-8a13-d36def478869)
But my testing now after your update started when the device got into sleeping mode. I'll check how things are tomorrow morning. 😊

**Comment by jhansche** [comment]
Yeah, let's leave it running for at least 70 minutes, and see if it updates wifi signal strength on its own. And check again tomorrow when it's no longer sleeping.
From what I can see right now on my own feeder (currently not sleeping as it is daytime), it is updating successfully now (but it was broken since Oct 19, before the fix).

If you're still seeing issues tomorrow during the day, search your logs for `birdbuddy` to see if any errors or warnings are appearing.

**Comment by RuudvanMunster** [comment]
It is working correctly! Thanx a lot for your support!
